### PR TITLE
bring back text generation webui

### PR DESCRIPTION
**Pull request by Butterroach**
implement this on the client side yourself im too lazy

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -24,8 +25,14 @@ def imgcaption(sid, data):
 
 
 @sio.event
-def chat(sid, data):
-    msg = send_message(send_message(data["message"]))
+def chat(sid, data: dict):
+    if data.get("textgenwui") is not None:
```
Comment (artifishvr) [inline main.py:29]: did you test this? idk how the client side would work

File: main.py [hunk 2]
```diff
@@ -24,8 +25,14 @@ def imgcaption(sid, data):
...
+        msg = textgenwui.send_message(
+            data["message"],
+            "Assistant",  # CHANGE THIS TO WHATEVER THE CHARACTER'S NAME IS!!!!
```
Comment (artifishvr) [inline main.py:32]: should be environment variable?

File: textgenwui.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
+# based on the chat example from textgen docs
+# i actually tested this
+# i forced myself to listen to lhugueny on loop while coding this so i can be more grateful for coding in silence
+
+import requests
+
+url = "http://127.0.0.1:5000/v1/chat/completions"
```
Comment (artifishvr) [inline textgenwui.py:7]: environment variables plssssss

File: textgenwui.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
+# based on the chat example from textgen docs
+# i actually tested this
```
Comment (artifishvr) [inline textgenwui.py:2]: poggers

**Review by artifishvr — CHANGES_REQUESTED** [review]
you should work on a branch of this repo instead so i can make these changes myself

File: main.py (hunk 1 above)
Comment (Butterroach) [reply main.py:29]: i am incapable of testing the main.py parts

File: main.py (hunk 2 above)
Comment (Butterroach) [reply main.py:32]: too lazy

File: textgenwui.py (hunk 3 above)
Comment (Butterroach) [reply textgenwui.py:7]: too lazy

File: main.py (hunk 2 above)
Comment (artifishvr) [reply main.py:32]: can you please move this to a branch of this repo so i can work on it then

File: textgenwui.py (hunk 3 above)
Comment (artifishvr) [reply textgenwui.py:7]: https://github.com/spongedsc/SpongeML/pull/8#discussion_r1574040804

File: main.py (hunk 2 above)
Comment (Butterroach) [reply main.py:32]: hold on

**Comment by artifishvr** [comment]
superseded by #9

**Pull request closed without merging**